src/services/localSSHService.ts

```diff
@@ -85,14 +99,18 @@ export class LocalSSHService extends Disposable implements ILocalSSHService {
     }
 
     private async copyProxyScript() {
-        const [proxyScript, launcher] = await Promise.all([
-            this.copyFileToGlobalStorage('out/local-ssh/proxy.js', 'sshproxy/proxy.js'),
-            isWindows ? this.copyFileToGlobalStorage('out/local-ssh/proxylauncher.bat', 'sshproxy/proxylauncher.bat') : this.copyFileToGlobalStorage('out/local-ssh/proxylauncher.sh', 'sshproxy/proxylauncher.sh')
-        ]);
-        if (!isWindows) {
-            await fsp.chmod(launcher, 0o755);
+        try {
+            const [proxyScript, launcher] = await Promise.all([
+                this.copyFileToGlobalStorage('out/local-ssh/proxy.js', 'sshproxy/proxy.js'),
+                isWindows ? this.copyFileToGlobalStorage('out/local-ssh/proxylauncher.bat', 'sshproxy/proxylauncher.bat') : this.copyFileToGlobalStorage('out/local-ssh/proxylauncher.sh', 'sshproxy/proxylauncher.sh')
+            ]);
+            if (!isWindows) {
+                await fsp.chmod(launcher, 0o755);
+            }
+            return { proxyScript, launcher };
+        } catch (e) {
+            throw new WrapError('Failed to copy local ssh proxy scripts', e, 'RaceCondition');
```

Comment:
I am not sure we should put here RaceCondition, why not propagate original error code? What if we are mistaken about race condition?

Reply:
Then we need a prefix or different event name. Because we can't identify if error comes from copy scripts or configure ssh config.

Most case copy scripts errors are not harmful (except first time to copy)